ciongke/cmd/hook/kubeagent.go

```diff
@@ -64,22 +64,33 @@ type kubeClient interface {
 // Cut off test-pr jobs after 10 hours.
 const jobDeadline = 10 * time.Hour
 
+func fields(kr KubeRequest) logrus.Fields {
```

Comment:
I don't think so. I'm not trying to change the behavior of `KubeRequest`, but instead writing a pure function that takes one and turns it into the other. Presumably I could have multiple such functions for different log levels or something, but I don't actually.